runelite-client/src/main/java/net/runelite/client/plugins/runenergy/RunEnergyPlugin.java

```diff
@@ -199,30 +210,42 @@ String getEstimatedRunTimeRemaining(boolean inSeconds)
 		}
 	}
 
-	private boolean isLocalPlayerWearingFullGraceful()
+	private int getGracefulRecoveryBoost()
 	{
 		final ItemContainer equipment = client.getItemContainer(InventoryID.EQUIPMENT);
 
 		if (equipment == null)
 		{
-			return false;
+			return 0;
 		}
 
 		final Item[] items = equipment.getItems();
 
-		// Check that the local player is wearing enough items to be using full Graceful
-		// (the Graceful boots will have the highest slot index in the worn set).
-		if (items == null || items.length <= EquipmentInventorySlot.BOOTS.getSlotIdx())
+		if (items == null)
+		{
+			return 0;
+		}
+
+		int boost = 0;
+
+		for (int i = 0; i < items.length; ++i)
+		{
+			final ImmutableSet<Integer> gracefulForSlot = GRACEFUL_EQUIPMENT_SLOT_MAP.get(i);
+
+			if (gracefulForSlot != null && gracefulForSlot.contains(items[i].getId()))
+			{
+				// 4% boost for the top and legs; 3% for the others
+				boost += i == EquipmentInventorySlot.BODY.getSlotIdx() || i == EquipmentInventorySlot.LEGS.getSlotIdx() ? 4 : 3;
```

Comment:
Instead of having hardcoded values here and below for the full set effect check, an enum with the slot index, set of graceful item IDs, and slot bonus could be defined which would give better definition to this data. This also would allow creation of some static `FULL_SET_BONUS` value that can be referenced below.